File: apple_orange.rb
# https://www.hackerrank.com/challenges/apple-and-orange/problem

def countApplesAndOranges(s, t, a, b, apples, oranges)
  puts apples.map { |apple| a + apple }
    .select { |apple| apple >= s && apple <= t }
    .count

  puts oranges.map { |orange| b + orange }
    .select { |orange| orange >= s && orange <= t }
    .count
end

File: apple_orange.py
# https://www.hackerrank.com/challenges/apple-and-orange/problem

def countApplesAndOranges(s, t, a, b, apples, oranges):
    apples = map(lambda x: a + x, apples)
    oranges = map(lambda x: b + x, oranges)
    apple_count = len(list(filter(lambda x: x >= s and x <= t, apples)))
    orange_count = len(list(filter(lambda x: x >= s and x <= t, oranges)))
    
    print(apple_count)
    print(orange_count)
